### PR TITLE
fix: adverse frame timestamps

### PR DESCRIPTION
**Pull request by armcknight**
Noticed that our profiling timestamps for slow/frozen frames all had the same value for start/end. The same code that should've executed after recording the timestamp was also before it, looks like a rebase gone wrong or something.

I also renamed a variable because the synonyms "last" and "previous" did not help to distinguish them.

### CODE REVIEW

File: Sources/Sentry/SentryFramesTracker.m [hunk 1]
```diff
@@ -134,28 +134,26 @@ - (void)displayLinkCallback
...
 #    if SENTRY_TARGET_PROFILING_SUPPORTED
-        [self recordTimestampStart:@(self.previousFrameTimestamp) end:@(lastFrameTimestamp)];
+        [self recordTimestampStart:@(self.previousFrameTimestamp) end:@(renderedFrameTimestamp)];
 #    endif // SENTRY_TARGET_PROFILING_SUPPORTED
     }
 
     atomic_fetch_add_explicit(&_totalFrames, 1, SentryFramesMemoryOrder);
-
-    self.previousFrameTimestamp = lastFrameTimestamp;
+    self.previousFrameTimestamp = renderedFrameTimestamp;
```
Comment (armcknight) [inline Sources/Sentry/SentryFramesTracker.m:156]: This was the problem. The same assignment to `self.previousFrameTimestamp` appeared both before and after recording the timestamps. It should only have been at the end.

Comment (kevinrenskers) [reply Sources/Sentry/SentryFramesTracker.m:156]: I noticed the same problem in one of my PRs but that got closed without getting merged. Should've copied the fix into a new PR 😓

But the fix seemed like a one-liner to me? See https://github.com/getsentry/sentry-cocoa/pull/2028/files#r934466813.

**Review by brustolin — APPROVED** [review]
LGTM

File: Sources/Sentry/SentryFramesTracker.m (hunk 1 above)
Comment (philipphofmann) [reply Sources/Sentry/SentryFramesTracker.m:156]: @armcknight, can you please add a test for this to avoid future bugs?

**Review by philipphofmann — COMMENTED** [review]
A test is missing and a changelog entry.

File: Sources/Sentry/SentryFramesTracker.m (hunk 1 above)
Comment (armcknight) [reply Sources/Sentry/SentryFramesTracker.m]: @kevinrenskers You're right, the only line needed to fix the bug was removing the assignment on line 142. The rest is code cleanup: renaming variables, preventing divide-by-0, etc.

**Comment by armcknight** [comment]
@philipphofmann check out the new tests here.

**Comment by armcknight** [comment]
The vlc integration test failure looks to be unrelated to these changes.

**Comment by philipphofmann** [comment]
> The vlc integration test failure looks to be unrelated to these changes.

Fixed here https://github.com/getsentry/sentry-cocoa/pull/2251 by @kevinrenskers.

File: CHANGELOG.md [hunk 2]
```diff
@@ -33,6 +33,7 @@
 - Use the `component` name source for SentryPerformanceTracker (#2168)
 - Add support for arm64 architecture to the device context (#2185)
 - Align core data span operations (#2222)
+- Fix profiling timestamps for slow/frozen frames (#2226)
```
Comment (philipphofmann) [inline CHANGELOG.md:36]: `m`: I think that needs to move up a bit.

**Review by philipphofmann — APPROVED** [review]
LGTM

**Review by kevinrenskers — APPROVED** [review]

**Pull request merged**